Support details report

stevegrunwell opened the pull request:
Following up on further discussion w/ @chrislema re: #32, he was looking for more of a "easily provide the user with any details they might need to provide to support", a la Easy Digital Downloads.

This PR extends the work done in #34 to include a plaintext report that a user can easily copy + paste into a support ticket, giving us verification of the Airstory requirements, as well as details about their environment, active plugins, theme, etc.

stevegrunwell committed:
Start populating a system report that can be submitted with support requests

stevegrunwell committed:
Continue the report, drawing more from EDD's awesome implementation. Also, ensure we're linking to EDD's repo

stevegrunwell committed:
Change label for web server details

bswatson commented on airstory.php:
```diff
@@ -16,6 +16,9 @@
 namespace Airstory;
 
+// Declare the canonical plugin version.
+define( 'AIRSTORY_VERSION', '0.1.0' );
```
I'm under the assumption this will be changed to 1.0.0 when it is released to the .org repo.

stevegrunwell replied:
Yeah, I typically start everything off as a 0.1.0 but with the stability and amount of testing in this I'm comfortable with a 1.0.0.

stevegrunwell replied:
#37 has been opened and the milestone updated to make sure this happens for release.

bswatson commented on includes/tools.php:
```diff
@@ -103,7 +103,10 @@ function render_tools_page() {
 
-
+		<h2><?php esc_html_e( 'Get Support', 'airstory' ); ?></h2>
+		<p><?php echo wp_kses_post( 'The Airstory WordPress plugin is backed by <a href="https://www.liquidweb.com/support/heroic-promise.html">Liquid Web\'s Heroic Support</a>; if you run into any issues when working with the Airstory plugin, please <a href="https://wordpress.org/support/plugin/airstory" target="_blank">feel free to open an issue in the WordPress.org plugin repo</a>.', 'airstory' ); ? ...
```
This feels like it should be wrapped in an `sprintf` to make the translation easier in the long term.

stevegrunwell replied:
A `sprintf()` or, you know, actually using a gettext function? Because I totally missed the latter, but not sure I agree on the former (especially since, for instance, the URL may change depending on the language, especially for the WordPress.org repo). Using the plugin in Spanish? Might as well link to the Spanish version of the repo, right?

bswatson commented on includes/tools.php:
```diff
@@ -223,3 +226,145 @@ function verify_https_support() {
+ * This is largely based on (a simplified version) of Easy Digital Downloads' edd_tools_sysinfo_get()
+ * reporting function.
+ *
+ * @link https://github.com/easydigitaldownloads/easy-digital-downloads
```
Would the inclusion of this as a derivative work change the license to GPL 2.0+?

stevegrunwell replied:
That's a really good question — I rewrote a fair portion of it (>50%, so it's more an "inspired by"), and it tears out all of the EDD-specific logic, but the pattern and formatting is still pretty damn similar. I'd like to avoid changing to GPL, if at all possible though, as the Airstory + Liquid Web teams were already in agreement on MIT (and MIT can be used more freely without the viral nature of the GPL).

At what point do you think we cross from being unique work to infringing on Pippin and team, thus invoking the GPL?

stevegrunwell replied:
Reached out to @pippinsplugins via Twitter DM to get his input, since he's the original copyright holder as owner of @easydigitaldownloads. I'd feel more comfortable getting his blessing before merging this + keeping MIT.

pippinsplugins replied:
I have no issue with this. :+1: from me.

stevegrunwell replied:
@pippinsplugins Thank you! Will note the link to this thread in the DocBlock for future reference.

![Phil Dunphy (of Modern Family) indicating that you're awesome](https://media.giphy.com/media/ClcWrARkrq1GM/giphy.gif)

stevegrunwell committed:
Merge branch 'develop' into feature/support-details

stevegrunwell committed:
Make the 'Get Support' section body i18n-ready on the Tools > Airstory page

stevegrunwell committed:
Merge branch 'develop' into feature/support-details

stevegrunwell committed:
Add link in get_support_details() DocBlock pointing to Pippin's approval to use EDD's reporting code as a template for our function while maintaining the MIT license.

Full discussion: https://github.com/liquidweb/airstory-wp/pull/35#discussion_r119370773
